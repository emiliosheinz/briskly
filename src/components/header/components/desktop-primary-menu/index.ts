export { DesktopPrimaryMenu } from './desktop-primary-menu.component'
