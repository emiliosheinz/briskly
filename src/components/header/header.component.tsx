import { signIn, signOut, useSession } from 'next-auth/react'
import Image from 'next/image'
import Link from 'next/link'

// TODO emiliosheinz: Create unit tests for Header component
// TODO emiliosheinz: Create Storybook documentation for Header component
export function Header() {
  const { data: sessionData } = useSession()

  const onClick = sessionData ? () => signOut() : () => signIn()
  const label = sessionData ? 'Sign out' : 'Sign in'

  function renderAuthButton() {
    return (
      <button
        className='hover:text-gray-900 whitespace-nowrap text-base font-medium text-primary-900'
        onClick={onClick}
      >
        {label}
      </button>
    )
  }

  return (
    <header className='relative bg-primary-50'>
      <div className='mx-auto max-w-7xl px-4 sm:px-6'>
        <div className='flex items-center justify-between border-b-2 border-primary-200 py-6 md:justify-start md:space-x-10'>
          <div className='flex justify-start lg:w-0 lg:flex-1'>
            <Link href='/'>
              <span className='sr-only'>Briskly</span>
              {/* TODO emiliosheinz: Change image component to use custom component */}
              <Image
                className='h-8 w-auto sm:h-10'
                src='/images/logo.png'
                width={80}
                height={80}
                alt='Briskly logo'
              />
            </Link>
          </div>
          <div className='flex flex-1 items-center justify-end lg:w-0'>
            {renderAuthButton()}
          </div>
        </div>
      </div>
    </header>
  )
}
