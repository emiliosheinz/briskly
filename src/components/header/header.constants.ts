import {
  ClockIcon,
  StarIcon,
  FireIcon,
  HeartIcon,
  PlusCircleIcon,
  UserCircleIcon,
  WrenchIcon,
} from '@heroicons/react/24/outline'

import { routes } from '~/utils/navigation'

const MENU_OPTIONS = [
  {
    label: 'Para Revisar',
    icon: ClockIcon,
    href: routes.toBeReviewed(),
    isAuthRequired: true,
  },
  {
    label: 'Mais Votados',
    icon: StarIcon,
    href: '#',
    isAuthRequired: false,
  },
  {
    label: 'Mais Recentes',
    icon: FireIcon,
    href: '#',
    isAuthRequired: false,
  },
  {
    label: 'Para Você',
    icon: HeartIcon,
    href: '#',
    isAuthRequired: true,
  },
  {
    label: 'Criar Deck',
    icon: PlusCircleIcon,
    href: routes.createNewDeck(),
    isAuthRequired: false,
  },
  {
    label: 'Meu Perfil',
    icon: UserCircleIcon,
    href: '#',
    isAuthRequired: true,
  },
  {
    label: 'Configurações',
    icon: WrenchIcon,
    href: '#',
    isAuthRequired: true,
  },
]

export const MOBILE_MENU_OPTIONS = [...MENU_OPTIONS]

export const DESKTOP_MENU_OPTIONS = {
  PRIMARY: MENU_OPTIONS.filter(({ isAuthRequired }) => !isAuthRequired),
  SECONDARY: MENU_OPTIONS.filter(({ isAuthRequired }) => isAuthRequired),
}
