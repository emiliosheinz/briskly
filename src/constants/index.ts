import { Visibility } from '@prisma/client'

import type { Option } from '~/components/radio-group'

export const MAX_TOPICS_PER_DECK = 5

export const DECK_VISIBILITY_OPTIONS: Array<Option<Visibility>> = [
  {
    name: 'Público',
    value: Visibility.Public,
    description: 'Todas as pessoas com acesso a este site',
  },
  {
    name: 'Privado',
    value: Visibility.Private,
    description: 'Apenas você e pessoas com acesso a sua conta',
  },
  {
    name: 'Apenas com link',
    value: Visibility.WithLink,
    description: 'Todos com acesso direto ao link do Deck',
  },
]

export const STUDY_SESSION_BOXES = [
  { reviewGapInHours: 24 },
  { reviewGapInHours: 24 * 3 },
  { reviewGapInHours: 24 * 7 },
  { reviewGapInHours: 24 * 15 },
  { reviewGapInHours: 24 * 20 },
]
