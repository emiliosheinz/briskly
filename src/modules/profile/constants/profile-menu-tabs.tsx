import dynamic from 'next/dynamic'

const DecksToBeReviewed = dynamic(() =>
  import('~/components/deck-card-list').then(
    module => module.DecksToBeReviewed,
  ),
)
const UserDecks = dynamic(() =>
  import('../components/user-decks.component').then(module => module.UserDecks),
)

const UserFavoriteDecks = dynamic(() =>
  import('~/components/deck-card-list').then(
    module => module.UserFavoriteDecks,
  ),
)

const DecksWithStudySession = dynamic(() =>
  import('../components/decks-with-study-session.component').then(
    module => module.DecksWithStudySession,
  ),
)

export const profileMenuTabs = [
  {
    name: 'Decks',
    content: UserDecks,
    isProfileOwnerOnly: false,
  },
  {
    name: 'Meus Estudos',
    content: DecksWithStudySession,
    isProfileOwnerOnly: true,
  },
  {
    name: 'Para Revisar',
    content: DecksToBeReviewed,
    isProfileOwnerOnly: true,
  },
  {
    name: 'Meus Favoritos',
    content: UserFavoriteDecks,
    isProfileOwnerOnly: true,
  },
]
