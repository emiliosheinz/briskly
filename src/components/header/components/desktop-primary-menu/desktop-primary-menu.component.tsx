import Link from 'next/link'

import { DESKTOP_MENU_OPTIONS } from '../../header.constants'

export function DesktopPrimaryMenu() {
  return (
    <div
      data-testid='desktop-primary-menu'
      className='mx-5 hidden gap-6 md:flex lg:mx-10 lg:gap-8'
    >
      {DESKTOP_MENU_OPTIONS.PRIMARY.map(option => (
        <Link
          href={option.href}
          key={option.label}
          className='flex items-center space-x-1 whitespace-nowrap text-primary-900'
        >
          {option.label}
        </Link>
      ))}
    </div>
  )
}
