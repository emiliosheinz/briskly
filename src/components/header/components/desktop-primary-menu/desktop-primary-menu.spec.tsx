import { render, screen } from '~/utils/tests/react-testing-library'

import { DESKTOP_MENU_OPTIONS } from '../../header.constants'
import { DesktopPrimaryMenu } from './desktop-primary-menu.component'

describe('Header DesktopPrimaryMenu Component', () => {
  it('should be correctly rendered', () => {
    const component = render(<DesktopPrimaryMenu />)

    expect(component.asFragment()).toMatchSnapshot()
  })

  it('should only be visible on desktop', async () => {
    render(<DesktopPrimaryMenu />)

    expect(screen.getByTestId('desktop-primary-menu')).toHaveClass('md:flex')
    expect(screen.getByTestId('desktop-primary-menu')).toHaveClass('hidden')
  })

  it('should render all DESKTOP_MENU_OPTIONS.PRIMARY menus', async () => {
    render(<DesktopPrimaryMenu />)

    for (const item of DESKTOP_MENU_OPTIONS.PRIMARY) {
      expect(screen.getByText(item.label)).toBeInTheDocument()
    }
  })
})
