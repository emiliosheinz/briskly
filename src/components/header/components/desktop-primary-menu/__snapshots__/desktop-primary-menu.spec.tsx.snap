// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header DesktopPrimaryMenu Component should be correctly rendered 1`] = `
<DocumentFragment>
  <div
    class="mx-5 hidden gap-6 md:flex lg:mx-10 lg:gap-8"
    data-testid="desktop-primary-menu"
  >
    <a
      class="flex items-center space-x-1 whitespace-nowrap text-primary-900"
      href="#"
    >
      Mais Votados
    </a>
    <a
      class="flex items-center space-x-1 whitespace-nowrap text-primary-900"
      href="#"
    >
      Mais Recentes
    </a>
    <a
      class="flex items-center space-x-1 whitespace-nowrap text-primary-900"
      href="#"
    >
      Criar Deck
    </a>
  </div>
</DocumentFragment>
`;
